lighthouse-core/report/v2/renderer/report-features.js

```diff
@@ -0,0 +1,326 @@
+    self.addEventListener('message', function msgHandler(/** @type {!Event} */ evt) {
+      const e = /** @type {!MessageEvent} */ (evt);
+
+      if (e.origin !== VIEWER_ORIGIN) {
+        return;
+      }
+
+      if (e.data['opened']) {
+        popup.postMessage({lhresults: this.json}, VIEWER_ORIGIN);
+        self.removeEventListener('message', msgHandler);
+      }
+    }.bind(this));
+
+    const popup = self.open(VIEWER_URL, '_blank');
+  }
+
+  /**
+   * Expands audit details when user prints via keyboard shortcut.
+   * @param {!Event} e
+   */
+  printShortCutDetect(e) {
+    if ((e.ctrlKey || e.metaKey) && e.keyCode === 80) { // Ctrl+P
+      this.expandAllDetails();
+    }
+  }
+
+  /**
+   * Expands all audit `<details>`.
+   * Ideally, a print stylesheet could take care of this, but CSS has no way to
+   * open a `<details>` element.
+   */
+  expandAllDetails() {
+    const details = this._dom.findAll('.lh-categories details', this._document);
+    details.map(detail => detail.open = true);
+  }
+
+  /**
+   * Collapses all audit `<details>`.
+   * open a `<details>` element.
+   */
+  collapseAllDetails() {
+    const details = this._dom.findAll('.lh-categories details', this._document);
+    details.map(detail => detail.open = false);
+  }
+
+  /**
+   * Sets up listeners to collapse audit `<details>` when the user closes the
+   * print dialog, all `<details>` are collapsed.
+   */
+  _setUpCollapseDetailsAfterPrinting() {
+    // FF and IE implement these old events.
+    if ('onbeforeprint' in self) {
+      self.addEventListener('afterprint', this.collapseAllDetails);
+    } else {
+      // Note: FF implements both window.onbeforeprint and media listeners. However,
+      // it doesn't matchMedia doesn't fire when matching 'print'.
+      self.matchMedia('print').addListener(mql => {
+        if (mql.matches) {
+          this.expandAllDetails();
+        } else {
+          this.collapseAllDetails();
+        }
+      });
+    }
+  }
+  /**
+   * Downloads a file (blob) using a[download].
+   * @param {!Blob|!File} blob The file to save.
+   */
+  _saveFile(blob) {
+    const filename = self.getFilenamePrefix({
+      url: this.json.url,
+      generatedTime: this.json.generatedTime
+    });
+
+    const ext = blob.type.match('json') ? '.json' : '.html';
+    const href = URL.createObjectURL(blob);
+
+    const a = this._dom.createElement('a');
+    a.download = `${filename}${ext}`;
+    a.href = href;
+    this._document.body.appendChild(a); // Firefox requires anchor to be in the DOM.
+    a.click();
+
+    // cleanup.
+    this._document.body.removeChild(a);
+    setTimeout(_ => URL.revokeObjectURL(href), 500);
+  }
+}
+
+if (typeof module !== 'undefined' && module.exports) {
+  module.exports = ReportUIFeatures;
+} else {
+  self.ReportUIFeatures = ReportUIFeatures;
+
+  /** @type {function({url: string, generatedTime: string}): string} */
+  self.getFilenamePrefix; // eslint-disable-line no-unused-expressions
```

Comment:
oh, I see the issue, `file-namer.js` needs to be added to the closure compiled files list (and should use this one's `{url: string, generatedTime: string}` instead of `Object` like in that file)